apluslms_roman/builder.py

```diff
@@ -84,14 +84,22 @@ def __init__(self, backend_class=None, settings=None):
         self._backend_class = backend_class
 
         name = getattr(backend_class, 'name', None) or backend_class.__name__
-        env_prefix = name.upper() + '_'
-        env = {k: v for k, v in environ.items() if k.startswith(env_prefix)}
+        prefix = name.upper() + '_'
+        options = {}
+        # config
         if backend_name is not None:
-            env.update({env_prefix + k.replace('-', '_').upper(): v
+            options.update({prefix + k.replace('-', '_').upper(): v
                 for k, v in settings['backends'][backend_name].items()
                 if k != 'type'})
+        # environment
+        options.update({k: v for k, v in environ.items() if k.startswith(prefix)})
+        # command line
+        if prefix == 'DOCKER_' and settings and 'docker' in settings:
+            options.update({prefix + k.replace('-', '_').upper(): v
+                for k, v in settings['docker'].items()})
```

Comment:
this could be made more general with `if prefix[:-1].lower() in settings`, would that work? will all backends follow that pattern?

Reply:
```
name = getattr(backend_class, 'name', None) or backend_class.__name__
prefix = name.upper() + '_'
```

Thus,

```
if name in settings
```

or actually

```
options = settings.get(name, {})
`´´

Reply:
or well the 'name' variable exists too, that would work better

Reply:
There are two other changes focusing this area. The multiple config per backend changes what dict is requested from settings (name of the config), but we still want the global env values based on backend base name (e.g. backend name is `my-docker` and we get settings from `my-docker`, but we load env keys `DOCKER_`

Reply:
yeah, that's exactly how it works currently?

Reply:
Apparently